### ISSUE TITLE
When evaluating local variables hangs, it cannot be canceled and Visual Studio hangs

### ISSUE THREAD

**Issue by abelbraaksma** [Microsoft/visualfsharp#4955]
I noticed this occasionally when debugging. The following popup shows:

![image](https://user-images.githubusercontent.com/16015770/40279167-02650f6e-5c3e-11e8-84ae-c02378741176.png)

It takes long, and you try to cancel it, but that has no effect. You can wait a while (sometimes a minute or so in my case) and it will unfreeze, but during that time, Visual Studio is inaccessible.

### Repro steps

No pure repro, unfortunately. On the same spot in my code, sometimes this happens, sometimes it is fast.  And when it is fast, it is not clear whether clicking cancel had effect or the evaluation just finished and dismissed the messagebox.

### Expected behavior

The messagebox should be cancelable.

### Actual behavior

Messagebox cannot be canceled and blocks Visual Studio.

### Known workarounds

Wait long enough, or kill Visual Studio.

### Related information 

Current Visual Studio 2017 Preview (15.7 Preview 6).

One extra observation, it turns out that in this particular location there are a lot of unnamed locals (leading to a not-too useful debugging experience as you have literally no idea what you are looking at, but that's another discussion), here's about a third of the list:

![image](https://user-images.githubusercontent.com/16015770/40279210-bae0ac4c-5c3e-11e8-81a7-9b849836827e.png)

**Comment by dsyme** [comment]
I don't think this is an F#-specific problem, and without a specific repro we're not going to be able to make much progress on either issue...

Repros are really needed per CONTRIBUTING.md so I'll close this - if you have one for either issue please reopen or file a new bug.

**Comment by abelbraaksma** [comment]
Yes, I anticipated an answer like that. Though I still think it's worthwhile reporting it. Issues that happen only once in a while are notoriously hard to repro but reporting it nonetheless may lead to others encountering the same issue chiming in. 

If I manage to find a repro I'll report back.

**Comment by robvon** [comment]
This issue occurs during debug once or twice a week on the latest VS update(s) (they come every other week) - and the only way out is to kill tasks under the main VS task until the whole IDE dies.
Most annoying. No predictable repro.

Really,  all such dialogs must be cancellable or else non-modal so that the primary window is left active and able to receive messages. Then maybe the debug can be stopped

Since the feature MOSTLY works I don't want to disable via options.

**Comment by nfdevil** [comment]
Since the 15.8.0 update, I'm having the same problem. With all my repos.

**Comment by cartermp** [comment]
@robvon and @nfdevil do you have a reproduction that you could share? If so, we'll gladly re-open this. We don't control the dialog, but if there is an issue with evaluation on our side we can see what we can do about fixing it.

**Comment by robvon** [comment]
Nope. It’s happening is quite non-deterministic.

 

Regards

 

Rob

**Comment by nfdevil** [comment]
I have uninstalled the 15.8.0 version and reinstalled it. This fixed it. Perhaps something happened in the update process

**Comment by eldenis** [comment]
![2022-09-06 00_47_18-XliffParser (Debugging) - Microsoft Visual Studio](https://user-images.githubusercontent.com/4098971/188556337-88aea6d4-7717-4887-816c-a0ee52d5d5b8.png)
It's happening for me on VS2022.

**Comment by abelbraaksma** [comment]
@eldenis interesting! Hasn't happened for me ever again on VS 2022. Maybe reopen as a new issue, considering how old this one is (and it's closed)? Hope you have a repo, it was something that I never managed to get.
